File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        pq = []

        for i in range(len(nums)):
            heapq.heappush(pq, -nums[i])
        
        f = k-1

        while(f > 0):
            heapq.heappop(pq)
            f -= 1
        return -pq[0]

File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.cpp
class Solution {
public:
    int findKthLargest(vector<int>& arr, int k) {
        priority_queue<int> pq;
        int n = arr.size();

        for(int i=0; i<n; i++) {
            pq.push(-arr[i]);
            if(pq.size() > k) {
                pq.pop();
            }
        }

        // for(int i=0; i<k; i++) {
        //     pq.push(-arr[i]);
        // }

        // for(int i=k; i<n; i++) {
        //     pq.push(-arr[i]);
        //     if(arr[i] >= pq.top()) {
        //         pq.pop();
        //     }
        // }

        return -pq.top();
    }
};